src/hooks/core/useCoreChatSystemHook.ts

```diff
@@ -159,13 +165,34 @@ const useCoreChatSystemHook = () => {
       );
       const chat = ChatSchema.fromWebsocketObject(websocketData);
       await chat.save(localDb);
-    } else {
-      const channelList = ChannelList.fromWebsocketObject(
-        websocketData,
-        channelType[websocketData?.channel_type]
-      );
-      await channelList.save(localDb);
+
+      return websocketData;
     }
+    /**
+     * TODO END
+     */
+
+    /**
+     * Process System Message Websocket
+     */
+    const newWebsocketData = {...websocketData};
+    const isSuccess = await saveSystemMessageFromWebsocket(
+      newWebsocketData?.message,
+      async (newMessage) => {
+        newWebsocketData.message = newMessage;
+        const channelList = ChannelList.fromWebsocketObject(
+          newWebsocketData,
+          channelType[websocketData?.channel_type]
+        );
+        const chat = ChatSchema.fromWebsocketObject(newWebsocketData);
+        await Promise.all([chat.save(localDb), channelList.save(localDb)]);
+      }
+    );
+
+    if (isSuccess) return websocketData;
+    /**
+     * Process System Message Websocket (END)
+     */
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of the `saveSystemMessageFromWebsocket` function call and related logic appears to be part of an effort to handle system messages in a more centralized manner. This could potentially improve the readability and maintainability of the code. However, it's important to ensure that this new approach doesn't introduce any unexpected side effects or bugs.
```diff
-    } else {
-      const channelList = ChannelList.fromWebsocketObject(
-        websocketData,
-        channelType[websocketData?.channel_type]
-      );
-      await channelList.save(localDb);
-     }
+    return websocketData;
+   }
+  /**
+   * TODO END
+   */
+
+  /**
+   * Process System Message Websocket
+   */
+  const newWebsocketData = {...websocketData};
+  const isSuccess = await saveSystemMessageFromWebsocket(
+    newWebsocketData?.message,
+    async (newMessage) => {
+      newWebsocketData.message = newMessage;
+      const channelList = ChannelList.fromWebsocketObject(
+        newWebsocketData,
+        channelType[websocketData?.channel_type]
+      );
+      const chat = ChatSchema.fromWebsocketObject(newWebsocketData);
+      await Promise.all([chat.save(localDb), channelList.save(localDb)]);
+    }
+  );
+
+  if (isSuccess) return websocketData;
+  /**
+   * Process System Message Websocket (END)
+   */
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->